ipn/ipnlocal/local.go

```diff
@@ -6009,16 +6017,36 @@ func (b *LocalBackend) setTCPPortsInterceptedFromNetmapAndPrefsLocked(prefs ipn.
 			b.updateServeTCPPortNetMapAddrListenersLocked(servePorts)
 		}
 	}
-	// Kick off a Hostinfo update to control if WireIngress changed.
+
+	// Kick off a Hostinfo update to control if ingress status has changed.
+	hostInfoChanged := false
 	if wire := b.wantIngressLocked(); b.hostinfo != nil && b.hostinfo.WireIngress != wire {
 		b.logf("Hostinfo.WireIngress changed to %v", wire)
 		b.hostinfo.WireIngress = wire
+		hostInfoChanged = true
+	}
+	if ie := b.hasIngressEnabledLocked(); b.hostinfo != nil && !eqBool(b.hostinfo.IngressEnabled, ie) {
+		b.logf("Hostinfo.IngressEnabled changed to %v", ie)
+		b.hostinfo.IngressEnabled.Set(ie)
+		hostInfoChanged = true
+	}
+	if hostInfoChanged {
 		b.goTracker.Go(b.doSetHostinfoFilterServices)
 	}
 
 	b.setTCPPortsIntercepted(handlePorts)
 }
 
+// eqBool compares boolean with a value stored in opt.Bool, returns true if they are equal or if opt.Bool is unset and
+// the boolean is false.
+func eqBool(ob opt.Bool, b bool) bool {
```

Comment:
I made the 'Hostinfo.IngressEnabled' field opt.Bool, so that control-side it can distinguish between it being unset because the client does not know that field vs being unset because it's not enabled, but then here we still want to treat unset as false

Reply:
alternatively, we can (should?) bump the current capability version to advertise that we even send this

then we don't have to send explicit "false" strings on the wire and can `omitempty` them instead and save bytes in the common case?

It'd be nice if we never sent "false" on the wire for either field, really. Maybe this is our chance to fix it.